torchao/prototype/mx_formats/custom_cast.py

```diff
@@ -1383,6 +1383,124 @@ def triton_to_mxfp8_dim1_reference(
             scale_e8m0_dim1,
         )
 
+    @triton.jit
+    def scale_swizzle(
```

Comment:
sgtm, maybe we can also rename the file to something like `kernels.py` since this isn't a cast?